### ISSUE TITLE
Use framework.ExpectNoError() for e2e tests

### ISSUE THREAD

**Issue by oomichi** [kubernetes/kubernetes#77103]
**What happened**:

After fixing golint failures under `test/e2e` (now still in progress), we need to put `gomega` for each `Expect()` and `HaveOccurred()`.
Then very common error checking code has become like:

`gomega.Expect(err).NotTo(gomega.HaveOccurred())`

We can replace this code with `framework.ExpectNoError(err)` which is more readable and easy to be understood what the code does.

Now we have following e2e test directories which can be replaced:
( - [x] shows done)
- [x] test/e2e (@oomichi https://github.com/kubernetes/kubernetes/pull/77693)
- [x] test/e2e/apimachinery (@draveness https://github.com/kubernetes/kubernetes/pull/77708)
- [x] test/e2e/apps (https://github.com/kubernetes/kubernetes/pull/77894)
- [x] test/e2e/auth (@oomichi)
- [x] test/e2e/autoscaling (@s-ito-ts)
- [x] test/e2e/common (@oomichi https://github.com/kubernetes/kubernetes/pull/78042)
- [x] test/e2e/framework/providers/gce
- [x] test/e2e/lifecycle (@atoato88 https://github.com/kubernetes/kubernetes/pull/77649)
- [x] test/e2e/lifecycle/bootstrap (@atoato88 https://github.com/kubernetes/kubernetes/pull/77754)
- [x] test/e2e/network (@k-toyoda-pi https://github.com/kubernetes/kubernetes/pull/77901 https://github.com/kubernetes/kubernetes/pull/77909)
- [x] test/e2e/node (@s-ito-ts https://github.com/kubernetes/kubernetes/pull/77902)
- [x] test/e2e/instrumentation (@draveness)
- [x] test/e2e/kubectl (@draveness)
- [x] test/e2e/scalability (@oomichi https://github.com/kubernetes/kubernetes/pull/77890)
- [x] test/e2e/scheduling (@oomichi https://github.com/kubernetes/kubernetes/pull/77890)
- [x] test/e2e/servicecatalog
- [x] e2e/storage/vsphere (https://github.com/kubernetes/kubernetes/pull/78021)
- [x] test/e2e/ui (@draveness)
- [x] test/e2e/upgrades/apps (@draveness https://github.com/kubernetes/kubernetes/pull/77627)
- [x] test/e2e/upgrades (@oomichi https://github.com/kubernetes/kubernetes/pull/77955)
- [x] test/e2e/upgrades/storage (@draveness)
- [x] test/e2e/windows (@draveness)

After finishing all the above, we need to add the checking script to the CI system for blocking PRs which contain this issue:

- [ ] https://github.com/kubernetes/kubernetes/pull/77612

**Comment by atoato88** [comment]
I can work to following items.
- test/e2e/lifecycle
- test/e2e/lifecycle/bootstrap

**Comment by k-toyoda-pi** [comment]
I think that we can replace `gomega.Expect(err).NotTo(gomega.HaveOccurred())`, too.
Files in the following directories have it. Could you add these directories to list?
- test/e2e/apimachinery
- test/e2e/apps
- test/e2e/instrumentation
- test/e2e/kubectl
- test/e2e/servicecatalog
- test/e2e/ui
- test/e2e/upgrades/apps
- test/e2e/upgrades
- test/e2e/upgrades/storage
- test/e2e/windows

**Comment by oomichi** [comment]
@k-toyoda-pi Thanks, I updated the list based on your comment :-)

**Comment by k-toyoda-pi** [comment]
Thanks, I can work to the following item.
- test/e2e/common

**Comment by s-ito-ts** [comment]
I can work to the following item.
- test/e2e/autoscaling

**Comment by draveness** [comment]
I'd like to take the files below.

```go
test/e2e/upgrades/apps
test/e2e/upgrades
test/e2e/upgrades/storage
```

**Comment by oomichi** [comment]
I am creating a hacking script which detects what code we need to update for this issue as https://github.com/kubernetes/kubernetes/pull/77612
Still we have almost 400 lines what we need to update at this time.

**Comment by draveness** [comment]
Here is the remaining files with `gomega.Expect(err)...`

```go
$ ag -l "gomega\.Expect\(err\)\.NotTo\(gomega.HaveOccurred\(\)\)" .
test/e2e/lifecycle/kubelet_security.go
test/e2e/lifecycle/reboot.go
test/e2e/lifecycle/restart.go
test/e2e/lifecycle/resize_nodes.go
test/e2e/common/pods.go
test/e2e/upgrades/storage/persistent_volumes.go
test/e2e/upgrades/storage/volume_mode.go
test/e2e/upgrades/apps/statefulset.go
test/e2e/upgrades/apps/job.go
test/e2e/apps/deployment.go
test/e2e/apps/daemon_set.go
```

I'd like to take files except for `test/e2e/apps` and `test/e2e/lifecycle` since they are already taken.

**Comment by draveness** [comment]
It seems like this issue will be solved after #77627 #77649 #77579 merged.

**Comment by oomichi** [comment]
@draveness 

`$ ag -l "gomega\.Expect\(err\)\.NotTo\(gomega.HaveOccurred\(\)\)"` which you are using cannot cover the following case which contains error message:
`gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to get pod %q", pod.Name)`
So still we have a lot of targets at this time :-(

**Comment by oomichi** [comment]
Thank you everyone who are fighting for this issue.
After the pull-kubernetes-verify job of https://github.com/kubernetes/kubernetes/pull/77612 pass, we can close this issue.

**Comment by oomichi** [comment]
Feel free to `/retest` on https://github.com/kubernetes/kubernetes/pull/77612 to verify our work status.

**Comment by draveness** [comment]
> @draveness
> 
> `$ ag -l "gomega\.Expect\(err\)\.NotTo\(gomega.HaveOccurred\(\)\)"` which you are using cannot cover the following case which contains error message:
> `gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to get pod %q", pod.Name)`
> So still we have a lot of targets at this time :-(

I just dug around to find the remaining case with the form of `gomega.Expect(err).NotTo(gomega.HaveOccurred(), "Failed to get pod %q", pod.Name)`:

```go
ag -l "gomega\.Expect\(err\)\.NotTo\(gomega.HaveOccurred\(\), "
test/e2e/servicecatalog/podpreset.go
test/e2e/lifecycle/addon_update.go
test/e2e/common/pods.go
test/e2e/windows/volumes.go
test/e2e/kubectl/portforward.go
test/e2e/apps/deployment.go
test/e2e/apps/daemon_set.go
test/e2e/apimachinery/crd_watch.go
test/e2e/apimachinery/webhook.go
test/e2e/apimachinery/table_conversion.go
test/e2e/apimachinery/etcd_failure.go
test/e2e/apimachinery/namespace.go
test/e2e/apimachinery/watch.go
test/e2e/apimachinery/garbage_collector.go
```

And I have left a message in https://github.com/kubernetes/kubernetes/pull/77649, the issue will be solved after:

+ https://github.com/kubernetes/kubernetes/pull/77649 update with the cmd I left and merged.
+ #77627 #77649 #77579 merged `gomega.Expect(err).NotTo(gomega.HaveOccurred())`
+ #77692 #77693 #77708  merged `gomega.Expect(err).NotTo(gomega.HaveOccurred(), "........")`

**Comment by draveness** [comment]
There are still codes with the form `gomega.Expect(errRW).NotTo(gomega.HaveOccurred(), msg)` when the var does not have name `err`, I could fix the problem after all the above PR get merged.

**Comment by atoato88** [comment]
done #77649.

**Comment by atoato88** [comment]
Maybe, following items are target of this issue too.
This is a case that is consisted of multiple lines.

```
test/e2e/windows/density.go:273:				30*time.Second, 10*time.Minute)).NotTo(gomega.HaveOccurred())
test/e2e/scheduling/limit_range.go:151:		})).NotTo(HaveOccurred())
test/e2e/scheduling/limit_range.go:193:		})).NotTo(HaveOccurred(), "kubelet never observed the termination notice")
test/e2e/common/pods.go:307:		})).NotTo(gomega.HaveOccurred(), "kubelet never observed the termination notice")
test/e2e/common/pods.go:808:			})).NotTo(gomega.HaveOccurred())
test/e2e/node/pods.go:162:			})).NotTo(HaveOccurred(), "kubelet never observed the termination notice")
test/e2e/node/kubelet_perf.go:79:	})).NotTo(HaveOccurred())
test/e2e/node/kubelet.go:327:				})).NotTo(HaveOccurred())
test/e2e/node/kubelet.go:333:					time.Second*30)).NotTo(HaveOccurred())
test/e2e/node/kubelet.go:349:					itArg.timeout)).NotTo(HaveOccurred())
test/e2e/apps/daemon_set.go:178:			NotTo(gomega.HaveOccurred(), "error waiting for daemon pod to not be running on nodes")
test/e2e/apps/daemon_set.go:240:			NotTo(gomega.HaveOccurred(), "error waiting for daemon pod to not be running on nodes")
test/e2e/apimachinery/etcd_failure.go:54:		})).NotTo(gomega.HaveOccurred())
```

These were gotten by `grep -n -e "NotTo(.*HaveOccurred"` and filtered in manual way.
It is needed that checks whether it is a target or not by reading each code, because these are just text match.

**Comment by draveness** [comment]
> Following items are target of this issue too.
> This is a case that is consisted of multiple lines.
> 
> ```
> test/e2e/windows/density.go:273:				30*time.Second, 10*time.Minute)).NotTo(gomega.HaveOccurred())
> test/e2e/scheduling/limit_range.go:151:		})).NotTo(HaveOccurred())
> test/e2e/scheduling/limit_range.go:193:		})).NotTo(HaveOccurred(), "kubelet never observed the termination notice")
> test/e2e/common/pods.go:307:		})).NotTo(gomega.HaveOccurred(), "kubelet never observed the termination notice")
> test/e2e/common/pods.go:808:			})).NotTo(gomega.HaveOccurred())
> test/e2e/node/pods.go:162:			})).NotTo(HaveOccurred(), "kubelet never observed the termination notice")
> test/e2e/node/kubelet_perf.go:79:	})).NotTo(HaveOccurred())
> test/e2e/node/kubelet.go:327:				})).NotTo(HaveOccurred())
> test/e2e/node/kubelet.go:333:					time.Second*30)).NotTo(HaveOccurred())
> test/e2e/node/kubelet.go:349:					itArg.timeout)).NotTo(HaveOccurred())
> test/e2e/apps/daemon_set.go:178:			NotTo(gomega.HaveOccurred(), "error waiting for daemon pod to not be running on nodes")
> test/e2e/apps/daemon_set.go:240:			NotTo(gomega.HaveOccurred(), "error waiting for daemon pod to not be running on nodes")
> test/e2e/apimachinery/etcd_failure.go:54:		})).NotTo(gomega.HaveOccurred())
> ```
> 
> These were gotten by `grep -n -e "NotTo(.*HaveOccurred"` and filtered in manual way.
> I've confirmed that these are target by checking each item.

Yeah, I noticed that and I'll fix all the remaining case after the PRs above get merged :)

**Comment by oomichi** [comment]
All PRs which try fixing this issue have been merged except my PRs at this time.
To make progress for merging https://github.com/kubernetes/kubernetes/pull/77612 I am making assignees free for each file which is not fixed now.
Please feel free to raise hands again if possible.

**Comment by k-toyoda-pi** [comment]
I can work to test/e2e/network.

**Comment by s-ito-ts** [comment]
I can work to test/e2e/node.

**Comment by oomichi** [comment]
I am working on test/e2e/scheduling and test/e2e/scalability.

**Comment by draveness** [comment]
I could take the files below 

```
test/e2e/apps
test/e2e/upgrades
test/e2e/upgrades/storage
test/e2e/upgrades/apps
```

**Comment by draveness** [comment]
This is the remaining file with regular format  `gomega.Expect(errRW).NotTo(gomega.HaveOccurred(), msg)` and `gomega.Expect(err).NotTo(gomega.HaveOccurred())`.

```
$ ag -l "gomega\.Expect\(err\)\.NotTo\(gomega.HaveOccurred\(\).*"
test/e2e/network/dns.go
test/e2e/network/firewall.go
test/e2e/network/proxy.go
test/e2e/network/service.go
test/e2e/network/ingress.go
test/e2e/network/dns_common.go
test/e2e/network/example_cluster_dns.go
test/e2e/network/network_tiers.go
test/e2e/network/network_policy.go
test/e2e/common/pods.go
test/e2e/node/kubelet.go
test/e2e/node/security_context.go
test/e2e/node/pre_stop.go
test/e2e/node/pods.go
test/e2e/node/ttlafterfinished.go
test/e2e/node/kubelet_perf.go
test/e2e/apps/disruption.go
```

 I'll take care of the `test/e2e/apps/disruption.go`, and the other files seem like already been taken.

**Comment by oomichi** [comment]
Thanks for doing this again @draveness 
`test/e2e/apps/disruption.go` had been fixed at once, but after that new test code contains `gomega.Expect(err).NotTo(gomega.HaveOccurred())` again.
So it is nice to take care of that again and now I really want to enable https://github.com/kubernetes/kubernetes/pull/77612 at the CI so that we don't need to take care of this issue forever.

**Comment by draveness** [comment]
> Thanks for doing this again @draveness
> `test/e2e/apps/disruption.go` had been fixed at once, but after that new test code contains `gomega.Expect(err).NotTo(gomega.HaveOccurred())` again.
> So it is nice to take care of that again and now I really want to enable #77612 at the CI so that we don't need to take care of this issue forever.

Yeah, that would be great, but we have to fix all the `gomega.Expect(err).NotTo(gomega.HaveOccurred())` ahead.

**Comment by k-toyoda-pi** [comment]
I've done all files in e2e/network at #77901 #77909.

**Comment by oomichi** [comment]
Thanks everyone, all PRs are ready for fixing this issue at this time.
I'd like to get attention from reviewers for making progress :-)

**Comment by k-toyoda-pi** [comment]
It seems that some files in e2e/common contain "Expect(err).NotTo(HaveOccurred()".
Is someone working to that? Can I take it?
- test/e2e/common/configmap_volume.go
- test/e2e/common/downwardapi_volume.go
- test/e2e/common/expansion.go
- test/e2e/common/init_container.go
- test/e2e/common/node_lease.go
- test/e2e/common/privileged.go
- test/e2e/common/projected_configmap.go
- test/e2e/common/projected_downwardapi.go
- test/e2e/common/projected_secret.go
- test/e2e/common/secrets_volume.go
- test/e2e/common/volumes.go

Oops, it might cause conflict between #77852 and this ...

**Comment by draveness** [comment]
> It seems that some files in e2e/common contain "Expect(err).NotTo(HaveOccurred()".
> Is someone working to that? Can I take it?
> 
> * test/e2e/common/configmap_volume.go
> * test/e2e/common/downwardapi_volume.go
> * test/e2e/common/expansion.go
> * test/e2e/common/init_container.go
> * test/e2e/common/node_lease.go
> * test/e2e/common/privileged.go
> * test/e2e/common/projected_configmap.go
> * test/e2e/common/projected_downwardapi.go
> * test/e2e/common/projected_secret.go
> * test/e2e/common/secrets_volume.go
> * test/e2e/common/volumes.go
> 
> Oops, it might cause conflict between #77852 and this ...

I'll work on this after the above PRs get merged.

**Comment by k-toyoda-pi** [comment]
I believe that `NotTo` is the same behavior as `ToNot`.
So, can we replace `Expect(err).ToNot(HaveOccurred())` to `ExpectNoError`?

**Comment by s-ito-ts** [comment]
It seems that "Expect(err).NotTo(gomega.HaveOccurred())" remains with some files under test/e2e/storage/vsphere.
I fix it in #78021

**Comment by oomichi** [comment]
@k-toyoda-pi 

> I believe that NotTo is the same behavior as ToNot.
> So, can we replace Expect(err).ToNot(HaveOccurred()) to ExpectNoError?

This is a really nice point, I also think they are the same.
However, it is better to concentrate on 'NotTo' case only at this time because that is majority and many people are still adding it. We need to put the CI script for blocking these additions with https://github.com/kubernetes/kubernetes/pull/77612
After that, we will be able to take care of 'ToNot' case with another issue.

**Comment by oomichi** [comment]
@s-ito-ts Thanks approved.

**Comment by oomichi** [comment]
@k-toyoda-pi 

> It seems that some files in e2e/common contain "Expect(err).NotTo(HaveOccurred()".
> Is someone working to that? Can I take it?

Sorry, that was my bad. I've taken care of that with the above PR.
I will try asking folks in KubeCon next week so that these all PRs are merged.

**Comment by odinuge** [comment]
Hi! Noticed that this only checked the normal e2e tests. I have made the script check `e2e_node` and `e2e_kubeadm` as well in https://github.com/kubernetes/kubernetes/pull/78779.
